CRM/Contribute/BAO/ManagePremiums.php

```diff
@@ -88,32 +82,47 @@ public static function setIsActive($id, $is_active) {
   /**
    * Add a premium product to the database, and return it.
    *
-   * @param array $params
-   *   Reference array contains the values submitted by the form.
-   * @param array $ids
-   *   Reference array contains the id.
+   * @param array $productParams
+   *   contains the values submitted by the form.
    *
    * @return CRM_Contribute_DAO_Product
    */
-  public static function add(&$params, &$ids) {
-    $params = array_merge(array(
-      'id' => CRM_Utils_Array::value('premium', $ids),
+  public static function add($productParams) {
+    $productParams = array_merge(array(
       'image' => '',
       'thumbnail' => '',
       'is_active' => 0,
       'is_deductible' => FALSE,
       'currency' => CRM_Core_Config::singleton()->defaultCurrency,
-    ), $params);
+    ), $productParams);
 
     // Modify the submitted values for 'image' and 'thumbnail' so that we use
     // local URLs for these images when possible.
-    $params['image'] = CRM_Utils_String::simplifyURL($params['image'], TRUE);
-    $params['thumbnail'] = CRM_Utils_String::simplifyURL($params['thumbnail'], TRUE);
+    $productParams['image'] = CRM_Utils_String::simplifyURL($productParams['image'], TRUE);
+    $productParams['thumbnail'] = CRM_Utils_String::simplifyURL($productParams['thumbnail'], TRUE);
 
-    // Save and return
+    // Save the new/updated premium product
     $premium = new CRM_Contribute_DAO_Product();
-    $premium->copyValues($params);
+    $premium->copyValues($productParams);
     $premium->save();
+
+    // If we updated a premium product we need to make sure we update the financial_type_id of that product in the
+    //   PremiumsProducts entity (which is used to record which products are available on each contribution page).
+    if (CRM_Utils_Array::value('id', $productParams)) {
+      $premiumProduct = new CRM_Contribute_DAO_PremiumsProduct();
+      $premiumProduct->product_id = $productParams['id'];
+      $premiumProduct->find();
+      while ($premiumProduct->fetch()) {
+        if (empty($params['financial_type_id'])) {
+          $premiumProduct->financial_type_id = 'null';
+        }
+        else {
+          $premiumProduct->financial_type_id = $params['financial_type_id'];
+        }
+        $premiumProduct->save();
+      }
+    }
+
```

Comment:
User is allowed to choose different FT for premium product through UI than FT assigned to actual product. I don't agree with the change here to forcefully update FT for premium product. Either we should display warning or error message when FT is changed.

Reply:
@pradpnayak I didn't totally understand how financial type was used in civicrm_premium_product and that it could be different to civicrm_premium. So I agree this change should not be in there - I've removed it.